src/classes/USER_UserService.cls

```diff
@@ -0,0 +1,271 @@
+      documentation and/or other materials provided with the distribution.
+    * Neither the name of Salesforce.org nor the names of
+      its contributors may be used to endorse or promote products derived
+      from this software without specific prior written permission.
+
+    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+    "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT 
+    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS 
+    FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE 
+    COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, 
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, 
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; 
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER 
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT 
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN 
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE 
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+
+/**
+* @author Salesforce.org
+* @date 2017
+* @group User
+* @description Service Class to handle actions on the User object.
+*/
+public with sharing class USER_UserService {
+
+	//================================
+	// Constants
+	//================================
+	public static final String SYSTEM_ADMINISTRATOR = 'System Administrator';
+	public static final String OBJECT_ID_PREFIX_USER = '005';
+	public static final String ERROR_NOTIFICATION_RECIPIENT_ALL_SYS_ADMINS = 'All Sys Admins';
+	public static final String JOB_TYPE_SCHEDULED_APEX = '7';
+	public static final String ERROR_TYPE_SCHEDULED_APEX = 'Scheduled Apex Error';
+	public static final String OBJECT_TYPE_USER = 'User';
+	public static final String ERROR_MESSAGE_SCHEDULED_APEX = 'There are scheduled jobs owned by an inactive or frozen user. Review your scheduled jobs and ensure the owner of each one is an active user.';
+	public static final String CONTEXT_TYPE_USER = 'User';
+	
+
+	/*********************************************************************************************************
+    * @description Determines if each User in the List is a Sys Admin based on ProfileId.
+    * @param updatedUsers The users that were modified.
+    * @return A Map containing the UserId and Boolean value based on the User's Profile.
+    **********************************************************************************************************/
+	public static Map<Id, Boolean> isSysAdmin(List<User> updatedUsers) {
+		Map<Id, Boolean> userIdToIsSysAdmin = new Map<Id, Boolean>();
+		
+		// Retrieve the SysAdmin Profile Id
+		Id sysAdminId = retrieveSysAdminProfileId();
+
+		// Determine if the User is a SysAdmin
+		if(null != updatedUsers && !updatedUsers.isEmpty()) {
+			for(User currentUser : updatedUsers) {
+				if(sysAdminId == currentUser.ProfileId) {
```

Comment:
you could code this up more efficiently with 

userIdToIsSysAdmin.put(currentUser.Id, sysAdminId == currentUser.ProfileId);

thus avoid the if/else, and the two separate invocations of put.

Reply:
Implemented this code change. Thanks!